auth0_flutter/lib/auth0_flutter_web.dart

```diff
@@ -1,24 +1,92 @@
 import 'package:auth0_flutter_platform_interface/auth0_flutter_platform_interface.dart';
 import 'src/version.dart';
 
+export 'package:auth0_flutter_platform_interface/auth0_flutter_platform_interface.dart'
+    show CacheLocation;
+
+/// Primary interface for interacting with Auth0 on web platforms.
 class Auth0Web {
   final Account _account;
 
   final UserAgent _userAgent =
       UserAgent(name: 'auth0-flutter', version: version);
 
+  /// Creates an instance of the [Auth0Web] client with the provided [domain]
+  /// and [clientId] properties.
+  ///
+  /// [domain] and [clientId] are both values that can be retrieved from the
+  /// application in your [Auth0 Dashboard](https://manage.auth0.com).
+  /// If you want to use your own implementation to handle credential storage,
   Auth0Web(final String domain, final String clientId)
       : _account = Account(domain, clientId);
 
-  Future<Credentials> onLoad() async {
-    await Auth0FlutterWebPlatform.instance.initialize(_account);
+  /// Initializes the client.
+  ///
+  /// This should be called during the loading phase of your application. If
+  /// the current user already has a session with Auth0, an instance of
+  /// [Credentials] will be returned, populated with their user data and tokens.
+  ///
+  /// Please see the [ClientOptions] type for the full description of the
+  /// available arguments to this method.
+  Future<Credentials> onLoad(
+      {final int? authorizeTimeoutInSeconds,
+      final CacheLocation? cacheLocation,
+      final String? cookieDomain,
+      final int? httpTimeoutInSeconds,
+      final String? issuer,
+      final int? leeway,
+      final bool? legacySameSiteCookie,
+      final int? sessionCheckExpiryDays,
+      final bool? useCookiesForTransactions,
+      final bool? useFormData,
+      final bool? useRefreshTokens,
+      final bool? useRefreshTokensFallback}) async {
+    await Auth0FlutterWebPlatform.instance.initialize(
```

Comment:
@Widcket I'm unsure as to whether it would be better just to take an instance of `ClientOptions` here rather than copy props over manually, thoughts?

Reply:
If this means that the developer would have to construct a `ClientOptions` just to pass to this method, I'd say no. It's convenient for us, but unnecessarily verbose for the developer, and kind of java-ish. Also would not be consistent with the rest of the public API.